processor/splitbatchprocessor/README.md

```diff
@@ -0,0 +1,30 @@
+# Split batch processor
+
+Splits a trace batch into multiple batches, one per trace.
+
+This processor will read the incoming batch and will call the next consumer once per trace: if a batch has 10 spans, each belonging to a different trace, the next consumer will be called 10 times.
+
+This is useful, or even required, to be used with components such as the Load Balancing Exporter. Future versions of the tail-based sampling processor might make use of this processor as well.
+
```

Comment:
Would this library be accepted as part of the core distribution?

Reply:
Is it needed in core by any of the components?

Reply:
> Is it needed in core by any of the components?

Not from core, but if we add a library to `-contrib`, then processors would have to depend both on `-contrib` and core. For components that don't live in contrib at all (private forks, for instance), it might be desirable to consume this from the core.

In any case: the same logic is required in three places today: groupbytrace processor, tail-based sampling, load balancer.

The tail-based sampling and the load balancer don't necessarily require this logic to be part of themselves, as they can rely on the groupbytrace processor to do this grouping. If we add the logic from this PR here as something that is available in the core's `pkg`, we'll potentially run the same logic twice: one in the groupbytrace processor, and one in the tail-based sampling/load balancer processors, which is why I thought that this processor here would make sense. With this processor, people can choose to use either groupbytrace or this processor here in front of their tail-based sampling/load balancer components.

Perhaps @chris-smith-zocdoc could share his views, as I know he's been using both the tail-based sampling and a (custom?) logic like this processor.

Reply:
>  it might be desirable to consume this from the core.

My build is already setup to reference exporters from `-contrib` so adding it to `-contrib` is fine with me. I think the most desirable aspect of core is discoverability, having some documentation that listed all the processors in both repositories would go a long way there.  

> which is why I thought that this processor here would make sense. With this processor, people can choose to use either groupbytrace or this processor here in front of their tail-based sampling/load balancer components.

My preference would be a library function that can be used by processors that need this functionality. I don't love having implicit dependencies between two processors, it would be quite difficult to troubleshoot why something isn't working if a required processor was missing from the pipeline. 

> we'll potentially run the same logic twice: one in the groupbytrace processor, and one in the tail-based sampling/load balancer processors

In many cases this would happen anyway as the load balancer + tail sampler would be different instances of the collector. If we're concerned about the performance impact, we can optimize the "no-op" case (preferably after writing some benchmarks)

Reply:
> as the load balancer + tail sampler would be different instances of the collector

The load balancer would only dispatch batches with single traces in it, so, there's no need for the tail-based sampling processor to do it again.

> If we're concerned about the performance impact, we can optimize the "no-op" case (preferably after writing some benchmarks)

True. Having a "don't split" flag would be a possible future solution if the performance hit isn't acceptable.

So, should this be a separate module here in contrib then?

Reply:
I would vote for that.

Reply:
Also for the moment can be in internal until someone externally needs to consume this library.

Reply:
I'll sign up as an external consumer of this library :) Right now I have this code copy-pasted into one of my processors